src/check_config_provider/redis_config_provider.rs

```diff
@@ -88,6 +138,94 @@ impl RedisConfigProvider {
             }
         }
     }
+
+    async fn monitor_updates(
+        &self,
+        manager: Arc<Manager>,
+        partitions: &[RedisPartition],
+        shutdown: CancellationToken,
+    ) {
+        let mut conn = self
+            .redis
+            .get_multiplexed_tokio_connection()
+            .await
+            .expect("Unable to connect to Redis");
+        let mut interval = interval(self.check_interval);
+
+        while !shutdown.is_cancelled() {
+            let _ = interval.tick().await;
+
+            for partition in partitions.iter() {
+                let mut pipe = redis::pipe();
+                // We fetch all updates from the list and then delete the key. We do this
+                // atomically so that there isn't any chance of a race
+                let (config_adds, config_deletes): (Vec<_>, Vec<_>) = pipe
+                    .atomic()
+                    .hvals(&partition.update_key)
+                    .del(&partition.update_key)
+                    .query_async::<(Vec<Vec<u8>>, ())>(&mut conn)
+                    .await
+                    .unwrap()
```

Comment:
Is the unwrap safe here?

Reply:
Probably not... I guess if we error, we should log an error and continue